File: DS In Java/QueuesInLinkedLists.java


public class QueuesInLinkedLists {
    private Node head;
    private Node front;
    private Node rear;
    class Node{
    private int value;
    private Node link;
    }
    QueuesInLinkedLists(int item){
        head=new Node();
        head.value=item;
        head.link=null;
        front=head;
        rear=head;
    }
    public void Enqueue(int item){
    Node n=new Node();
    n.value=item;
    n.link=null;
    rear.link=n;
    rear=n;
    }
    public void Print(){
    Node z=front;
    while(z!=null){
    System.out.print(z.value);
    z=z.link;
    }
    System.out.println();
    }
    public void Dequeue(){
    Node n=front;
    front=n.link;
    }
    public static void main(String[] args){
    QueuesInLinkedLists list=new QueuesInLinkedLists(3);
    System.out.println("Before Dequeue Operation");
    list.Print();
    list.Enqueue(4); list.Print();
    list.Enqueue(2); list.Print();
    list.Enqueue(1); list.Print();
    list.Enqueue(5); list.Print();
    list.Enqueue(7); list.Print();
    list.Dequeue();
    System.out.println("After Dequeue Operation :");
    list.Print();}
}

File: DS Implementations/QueuesInLinkedLists.c
#include <stdio.h>
#include <stdlib.h>
struct Node{
	int data;
	struct Node* next;
	};
	struct Node* front;
	struct Node* rear;
void Enqueue(int x){
	struct Node* temp=malloc(sizeof(struct Node));
	temp->data=x;
	temp->next=NULL;
	if(front==NULL){
	front=temp;
	rear=temp;
	return;
}
	rear->next=temp;
	rear=temp;
}
void Dequeue(){
	struct Node* temp=front;
	if(front==NULL){
		return;
	}	
	if(front==rear){
		front=rear=NULL;
		}
	else
	front=temp->next;
	free(temp);
	}
	void Print(){
		struct Node* temp=front;
		printf("Queue : ");
		while(temp!=NULL){
			printf("%d ",temp->data);
			temp=temp->next;
			}
			printf("\n");
		}
int main(){
	Enqueue(2); Print(); 
	Enqueue(4); Print();
	Enqueue(6); Print();
	Dequeue();  Print();
	Enqueue(8); Print();
	return 0;
}
